### PR TITLE
Use coinselect for finding utxos

### PR DESCRIPTION
**Pull request by harshjv**
- [x] Use https://github.com/bitcoinjs/coinselect for calculating change & utxos
- [x] Move `calculateFee` to BitcoinUtil

### CODE REVIEW

File: src/providers/bitcoin/BitcoinLedgerProvider.js [hunk 1]
```diff
@@ -142,7 +131,49 @@ export default class BitcoinLedgerProvider extends LedgerProvider {
...
+          const addr = find(addrList, addr => addr.address === utxo.address)
+
+          utxo.value = utxo.satoshis
+          utxo.derivationPath = addr.derivationPath
+
+          return utxo
+        })
+
+      globalUtxoSet = globalUtxoSet.concat(utxos)
+      if (feePerByte === false) feePerByte = await feePerBytePromise
```
Comment (monokh) [inline src/providers/bitcoin/BitcoinLedgerProvider.js:159]: confused why the promise isn't just called here instead of using a variable?

**Review by monokh — APPROVED** [review]
LGTM! minor comment
Just checking you have manage to run the integration tests to ensure the ledger swaps continue to work?

File: src/providers/bitcoin/BitcoinLedgerProvider.js (hunk 1 above)
Comment (harshjv) [reply src/providers/bitcoin/BitcoinLedgerProvider.js:159]: We are initiating `getFeePerByte` on line 107 and after that we are also making other network calls. While `getFeePerByte` isn't required until line 161, it is essentially being called in parallel in a non-blocking way.

**Comment by harshjv** [comment]
@monokh all green!

**Pull request merged**